### ISSUE TITLE
Cannot run `getting-started` example with k3s

### ISSUE THREAD

**Issue by KoltesDigital** [GoogleContainerTools/skaffold#8022]
Following #8021 I tried to run again the `getting-started` example, which did work some days ago, and now does not. Inbetween I've updated the binaries (helm / skaffold / ...), and I don't recall changing k8s / k3s / ... configs.

### Expected behavior

It should "work".

### Actual behavior

Docker image cannot be instantiated, because it doesn't seem to exist.

Note that I'm using k3s. Skaffold should detect that the cluster is local. However in practice I don't know what that means.

The image is built on the Docker deamon outside the k3d cluster.

### Information

- Skaffold version: **v2.0.1**
- Operating system: **Ubuntu 22.04 on WSL2 Win10 22H2**
- Installed via: `curl -Lso skaffold https://storage.googleapis.com/skaffold/releases/latest/skaffold-linux-amd64 && install skaffold /usr/local/bin/`
- Contents of skaffold.yaml: https://github.com/GoogleContainerTools/skaffold/blob/main/examples/getting-started/skaffold.yaml

### Steps to reproduce the behavior

```
git clone https://github.com/GoogleContainerTools/skaffold.git
cd skaffold/examples/getting-started
skaffold dev
```

```
Listing files to watch...
 - skaffold-example
Generating tags...
 - skaffold-example -> skaffold-example:v2.0.1-5-g3ac63ca3b
Checking cache...
 - skaffold-example: Found Locally
Tags used in deployment:
 - skaffold-example -> skaffold-example:c74b973e42c15d4816507925e8b47b1c482babe3c5bd922fd3f2d7631b3a226c
Starting deploy...
Loading images into k3d cluster nodes...
Images loaded in 101ns
 - pod/getting-started created
Waiting for deployments to stabilize...
 - pods: container getting-started is waiting to start: skaffold-example:c74b973e42c15d4816507925e8b47b1c482babe3c5bd922fd3f2d7631b3a226c can't be pulled
    - pod/getting-started: container getting-started is waiting to start: skaffold-example:c74b973e42c15d4816507925e8b47b1c482babe3c5bd922fd3f2d7631b3a226c can't be pulled
 - pods failed. Error: container getting-started is waiting to start: skaffold-example:c74b973e42c15d4816507925e8b47b1c482babe3c5bd922fd3f2d7631b3a226c can't be pulled.
Cleaning up...
 - pod "getting-started" deleted
1/1 deployment(s) failed
```

**Comment by ericzzzzzzz** [comment]
may be the same cause as #7992

**Comment by aaron-prindle** [comment]
This was related to a recent regression where Skaffold would not properly loading images into k3d and kind clusters.  This root cause here is a dupe of #7992, follow the issue there for more info/discussion.  This should be resolved now @ HEAD with #8007 and will be available in our next release to be released later this week
